### PR TITLE
Onboard Hypershift

### PR DESCRIPTION
**Pull request by zkayyali812**
Signed-off-by: Zachary Kayyali <zkayyali@redhat.com>
https://github.com/stolostron/backlog/issues/20258

### CODE REVIEW

**Review by ianzhang366 — APPROVED** [review]

File: pkg/templates/crds/hypershift/cluster.open-cluster-management.io_hypershiftdeployments.yaml [hunk 1]
```diff
@@ -0,0 +1,7235 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (zhujian7) [inline pkg/templates/crds/hypershift/cluster.open-cluster-management.io_hypershiftdeployments.yaml:1]: Could you please use the latest version of crd? I think there are some updates. https://github.com/stolostron/hypershift-deployment-controller/blob/main/config/crd/cluster.open-cluster-management.io_hypershiftdeployments.yaml

File: pkg/templates/charts/toggle/hypershift/templates/hypershiftDeployment_clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  resources:
+  - manifestworks
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
```
Comment (zhujian7) [inline pkg/templates/charts/toggle/hypershift/templates/hypershiftDeployment_clusterrole.yaml:76]: could you please add this permission? https://github.com/stolostron/hypershift-addon-chart/pull/5/files  thanks

Comment (zhujian7) [inline pkg/templates/charts/toggle/hypershift/templates/hypershiftDeployment_clusterrole.yaml:76]: could you please add this permission? https://github.com/stolostron/hypershift-addon-chart/pull/5/files

File: pkg/templates/charts/toggle/hypershift/templates/hypershift-addon-manager_clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Values.org }}:{{ .Chart.Name }}:hypershift-addon-manager
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  - configmaps
```
Comment (ianzhang366) [inline pkg/templates/charts/toggle/hypershift/templates/hypershift-addon-manager_clusterrole.yaml:10]: @zkayyali812 can you please add `events` over here

Comment (zkayyali812) [reply pkg/templates/charts/toggle/hypershift/templates/hypershift-addon-manager_clusterrole.yaml]: updated

File: pkg/templates/crds/hypershift/cluster.open-cluster-management.io_hypershiftdeployments.yaml (hunk 1 above)
Comment (zkayyali812) [reply pkg/templates/crds/hypershift/cluster.open-cluster-management.io_hypershiftdeployments.yaml:3]: updated

**Comment by zhujian7** [comment]
@zkayyali812 @ianzhang366 Can we cancel the hold now？

**Review by JakobGray — APPROVED** [review]

**Comment by zkayyali812** [comment]
Going to merge PR now.
Component still requires further updates and is not completely working as expected yet.

We are requiring further changes to ensure that all image references are stored and managed in the hypershift deployment templates

**Pull request merged**